test/org/sagebionetworks/bridge/cache/CacheProviderMigrationTest.java

```diff
@@ -0,0 +1,387 @@
+        
+        verify(newJedisOps).get("sessionToken:session");
+        verify(oldJedisOps).get("sessionToken:session");
+    }
+    
+    @Test
+    public void getUserSessionNewCache() throws Exception {
+        UserSession session = stubSession("userId");
+        when(newJedisOps.get("sessionToken:session")).thenReturn(MAPPER.writeValueAsString(session));
+        
+        UserSession returned = cacheProvider.getUserSession("sessionToken");
+        assertEquals("userId", returned.getId());
+        
+        verify(newJedisOps).get("sessionToken:session");
+        verify(oldJedisOps, never()).get("sessionToken:session");
+    }
+    
+    @Test
+    public void getUserSessionByUserIdNoCache() {
+        UserSession returned = cacheProvider.getUserSessionByUserId("userId");
+        assertNull(returned);
+        
+        verify(newJedisOps).get("userId:session:user");
+        verify(oldJedisOps).get("userId:session:user");
+    }
+
+    @Test
+    public void getUserSessionByUserIdOldCache() throws Exception {
+        UserSession session = stubSession("userId");
+        when(oldJedisOps.get("sessionToken:session")).thenReturn(MAPPER.writeValueAsString(session));
+        when(oldJedisOps.get("userId:session:user")).thenReturn("sessionToken");
+        
+        UserSession returned = cacheProvider.getUserSessionByUserId("userId");
+        assertEquals("userId", returned.getId());
+        
+        verify(newJedisOps).get("userId:session:user");
+        verify(oldJedisOps).get("userId:session:user");
+    }
+
+    @Test
+    public void getUserSessionByUserIdNewCache() throws Exception {
+        UserSession session = stubSession("userId");
+        when(newJedisOps.get("sessionToken:session")).thenReturn(MAPPER.writeValueAsString(session));
+        when(newJedisOps.get("userId:session:user")).thenReturn("sessionToken");
+
+        UserSession returned = cacheProvider.getUserSessionByUserId("userId");
+        assertEquals("userId", returned.getId());
+        
+        verify(newJedisOps).get("userId:session:user");
+        verify(oldJedisOps, never()).get("userId:session:user");
+    }
+
+    @Test
+    public void removeSession() {
+        mockSessionAndTransaction(newJedisOps);
+        UserSession session = mockSessionAndTransaction(oldJedisOps);
+        
+        cacheProvider.removeSession(session);
+        
+        verify(transaction, times(2)).del("sessionToken:session");
+        verify(transaction, times(2)).del("userId:session:user");
+    }
+
+    @Test
+    public void removeSessionByUserId() {
+        mockSessionAndTransaction(newJedisOps);
+        mockSessionAndTransaction(oldJedisOps);
+        
+        when(newJedisOps.get("userId:session:user")).thenReturn("sessionToken");
+        
+        cacheProvider.removeSessionByUserId("userId");
+        
+        verify(transaction, times(2)).del("sessionToken:session");
+        verify(transaction, times(2)).del("userId:session:user");
+    }
```

Comment:
It doesn't really matter since it's harmless to call delete if there's no session. We just need to be sure we call delete on both Redis instances.